code/concierge/src/main/kotlin/com/adobe/marketing/mobile/concierge/ui/components/footer/ChatFooter.kt

```diff
@@ -34,41 +34,57 @@ import com.adobe.marketing.mobile.concierge.ui.state.FeedbackEvent
  * @param citations List of [Citation] to display in the sources accordion.
  * @param interactionId interaction ID for feedback buttons.
  * @param onFeedback Callback invoked when a feedback button is pressed.
+ * @param feedbackState Current feedback state for this interaction.
  */
 @Composable
 internal fun ChatFooter(
     modifier: Modifier = Modifier,
-    citations: List<Citation>,
+    citations: List<Citation>?,
     interactionId: String?,
-    onFeedback: (FeedbackEvent) -> Unit
+    onFeedback: (FeedbackEvent) -> Unit,
+    feedbackState: FeedbackState = FeedbackState.None
 ) {
+    val hasCitations = !citations.isNullOrEmpty()
+    val hasInteractionId = !interactionId.isNullOrEmpty()
+    
+    // Early return if no sources to show or if unable to send feedback due to no interaction id
+    if (!hasCitations && !hasInteractionId) return
```

Comment:
For composables, I recommend avoiding return and gate them in clauses from the parent component. While this is OK, for cases where a previously rendered item has to be removed from the screen, early returns don't play well with compose compiler.